components/service/glean/docs/metrics/adding-new-metrics.md

```diff
@@ -246,5 +417,195 @@ assertEquals(11, snapshot.sum)
 
 // Usually you don't know the exact timing values, but how many should have been recorded.
 assertEquals(2L, snapshot.count())
+```
+
+## Datetimes
+
+Datetimes are used to record an absolute date and time, for example the date and time that the application was first run.
+
+The device's offset from UTC is recorded and sent with the datetime value in the ping.
+
+### Metric-specific properties
+
+Datetimes have a required `time_unit` parameter to specify the smallest unit
+of resolution that the timespan will record. The allowed values for `time_unit` are:
+
+   `nanosecond`, `microsecond`, `millisecond`, `second`, `minute`, `hour`, and
+   `day`
+
+Carefully consider the required resolution for recording your metric, and choose
+the coarsest resolution possible.
+
+### Usage
+
+You first need to add an entry for it to the `metrics.yaml` file:
+
+```YAML
+install:
+  first_run:
+    type: datetime 
+    time_unit: day 
+    description: >
+      Records the date when the application was first run
+    ...
+```
+
+Now that the datetime is defined in `metrics.yaml` you can use the metric to
+record datetimes in the application's code:
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Install
+
+Install.firstRun.set() // Records "now"
+Install.firstRun.set(Calendar(2019, 3, 25)) // Records a custom datetime
+```
+
+There are test APIs available too.
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Install
+Glean.enableTestingMode()
+
+// Was anything recorded?
+assertTrue(Install.firstRun.testHasValue())
+// Was it the expected value? 
+assertEquals("2019-03-25-05:00", Install.firstRun.testGetValueAsString())
```

Comment:
nit: Your defined metric above has a time unit of `day` and this is comparing as if it were `hour`

Reply:
The `-05:00` bit is actually the timezone offset.  If there were a time, it would look like `2019-03-25T09:15:00-05:00`.  I agree this is confusing -- maybe I'll add a comment to this effect.

Reply:
Doh!  Makes sense now that I look at it with that in mind ;)